### PR TITLE
added authorization and authentication

### PR DESCRIPTION
**Pull request by Andrei-Suta**

### CODE REVIEW

**Review by mgandore — APPROVED** [review]
Looks fine to me.
* I noticed that in the last version of to-do-api repo it's created a special endpoint for changing task status. (rn we are using the same endpoint as for editing a task).
** Differences regarding last three commits are on me, as I did not change indentation to those files. Did not thought it's necessary to do so.

File: to-do-ui/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { NgModule } from "@angular/core";
+import { RouterModule, Routes } from "@angular/router";
+
+import { LoginComponent } from "./components/login/login.component";
+import { RegisterComponent } from "./components/register/register.component";
+import { TaskListComponent } from "./components/task-list/task-list.component";
+
+
```
Comment (danielursu) [inline to-do-ui/src/app/app-routing.module.ts:8]:
```suggestion
```

File: to-do-ui/src/app/app.component.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
-import { Component } from "@angular/core";
+import { AfterContentInit, Component, OnInit, TemplateRef, ViewChild, ViewContainerRef } from "@angular/core";
```
Comment (danielursu) [inline to-do-ui/src/app/app.component.ts:1]: delete unused imports

File: to-do-ui/src/app/auth/auth.module.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { InputTextareaModule } from "primeng/inputtextarea";
+import { InputTextModule } from "primeng/inputtext";
+
+
+
+@NgModule({
+	declarations: [RegisterComponent, LoginComponent],
+	imports: [
+		CommonModule,
+		MenubarModule,
```
Comment (danielursu) [inline to-do-ui/src/app/auth/auth.module.ts:18]: MenuBarModule is also imported in AppModule. Try to delete this one

File: to-do-ui/src/app/auth/auth.module.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { NgModule } from "@angular/core";
+import { CommonModule } from "@angular/common";
+
+import { RegisterComponent } from "../components/register/register.component";
+import { LoginComponent } from "../components/login/login.component";
+import { MenubarModule } from "primeng/menubar";
+import { FormsModule } from "@angular/forms";
+import { ButtonModule } from "primeng/button";
+import { InputTextareaModule } from "primeng/inputtextarea";
+import { InputTextModule } from "primeng/inputtext";
```
Comment (danielursu) [inline to-do-ui/src/app/auth/auth.module.ts:10]: move these lines to the top of the imports

File: to-do-ui/src/app/components/navbar/navbar.component.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import { LocalStorageService } from "src/app/shared/localStorage.service";
+
+
+@Component({
+	selector: "navbar-component",
+	templateUrl: "./navbar.component.html",
+	styleUrls: ["./navbar.component.css"]
+})
+export class NavbarComponent implements OnInit {
+	public authenticationState?: boolean;
```
Comment (danielursu) [inline to-do-ui/src/app/components/navbar/navbar.component.ts:12]:
```suggestion
	public isUserLoggedIn: boolean = false;
```

File: to-do-ui/src/app/components/navbar/navbar.component.html [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<div class="container" *ngIf="!this.authenticationState">
+	<button pButton type="button" label="Tasks" class="p-button-outlined" routerLink="tasks"></button>
+	<button pButton type="button" label="Logout" class="p-button-outlined" (click)="this.handleLogOut()"></button>
+</div>
+
+<div class="container" *ngIf="this.authenticationState">
```
Comment (danielursu) [inline to-do-ui/src/app/components/navbar/navbar.component.html:6]:
```suggestion
<div class="container" *ngIf="!this.isUserLoggedIn">
```

File: to-do-ui/src/app/components/navbar/navbar.component.html [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<div class="container" *ngIf="!this.authenticationState">
```
Comment (danielursu) [inline to-do-ui/src/app/components/navbar/navbar.component.html:1]:
```suggestion
<div class="container" *ngIf="this.isUserLoggedIn">
```

File: to-do-ui/src/app/components/register/register.component.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+})
+export class RegisterComponent {
+	public password?: string;
+	public email?: string;
+	public name?: string;
+	public response?: any;
+
+	public constructor(
+		private authService: AuthService,
+		private router: Router) { }
```
Comment (danielursu) [inline to-do-ui/src/app/components/register/register.component.ts:20]:
```suggestion
	public constructor(
		private authService: AuthService,
		private router: Router
	) { }
```

File: to-do-ui/src/app/components/register/register.component.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+@Component({
+	selector: "register-component",
+	templateUrl: "./register.component.html",
+	styleUrls: ["./register.component.css"]
+})
+export class RegisterComponent {
+	public password?: string;
+	public email?: string;
+	public name?: string;
```
Comment (danielursu) [inline to-do-ui/src/app/components/register/register.component.ts:15]: registerDTO

File: to-do-ui/src/app/components/register/register.component.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+	public response?: any;
+
+	public constructor(
+		private authService: AuthService,
+		private router: Router) { }
+
+	public handleRegister(): void {
+		const registerDTO: RegisterDTO = { email: this.email, password: this.password, name: this.name };
+		this.response = this.authService.registerUser(registerDTO).subscribe(res => console.log(res));
```
Comment (danielursu) [inline to-do-ui/src/app/components/register/register.component.ts:24]: do not store response in this way
if you want the response try .subscribe(res => this.response = res)
add a //TODO comment in order to add toasts

File: to-do-ui/src/app/components/register/register.component.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+		this.router.navigate(["/login"]);
```
Comment (danielursu) [inline to-do-ui/src/app/components/register/register.component.ts:25]: move this line inside subscribe handler
you want to navigate after the registration

File: to-do-ui/src/app/shared/api-caller.service.ts [hunk 12]
```diff
@@ -8,29 +8,46 @@ import { TaskDTO } from "./task-dto";
...
+		});
 	}
 
 	public deleteTask(id: number): Observable<object> {
-		return this.http.patch(`${this.url}/${id}`, { id });
+		return this.http.patch(`${this.url}/delete/${id}`, { id }, {
+			headers: { "Authorization": "Bearer " + localStorage.getItem("Token"), "Content-Type": "application/json" }
+		});
+
+
```
Comment (danielursu) [inline to-do-ui/src/app/shared/api-caller.service.ts:50]: emtpy lines

File: to-do-ui/src/app/components/login/login.component.ts [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+@Component({
+	selector: "login-component",
+	templateUrl: "./login.component.html",
+	styleUrls: ["./login.component.css"]
+})
+export class LoginComponent {
+	public password?: string;
+	public email?: string;
+	public name?: string;
```
Comment (danielursu) [inline to-do-ui/src/app/components/login/login.component.ts:17]: loginDTO

**Review by danielursu — CHANGES_REQUESTED** [review]

**Pull request merged**